lib/assets/javascripts/turbograft/turbolinks.coffee

```diff
@@ -307,12 +314,11 @@ class window.Turbolinks
       Turbolinks.pushState { turbolinks: true, url: url.absolute }, '', url.absolute
     return
 
-  reflectRedirectedUrl = (xhr) ->
-    if location = xhr.getResponseHeader 'X-XHR-Redirected-To'
-      location = new ComponentUrl location
-      preservedHash = if location.hasNoHash() then activeDocument.location.hash else ''
-      Turbolinks.replaceState currentState, '', location.href + preservedHash
-
+  reflectRedirectedUrl = (response) ->
+    if url = response.redirectedTo()
```

Comment:
Aha, here's another use of that method. Maybe it should just be a property, since it can be determined in `Response`'s constructor?